test/benchmarks/README.md

```diff
@@ -17,3 +16,32 @@ To filter out which benchmarks are performed use:
 ```
 --benchmark_filter="adler32*"
 ```
+
+There are two different benchmarks, micro and macro.
+
+### Benchmark benchmark_zlib
+These are microbenchmarks intended to test lower level subfunctions of the library.
+
+Benchmarks include impelementations of:
+    - Adler32
+    - CRC
+    - 256 byte comparisons
+    - SIMD accelerated "slide hash" routine
+
+By default these benchmarks report things on the nanosecond scale and are small enough
+to measure very minute diferences.
+
+### Benchmark benchmark_zlib_apps
+These benchmarks measure applications of zlib as a whole.  Currently the only examples
+are PNG encoding and decoding. The PNG encode and decode tests leveraging procedurally
+generated and highly compressible image data.
+
+Additionally, a test called `png_decode_realistic` that will decode any RGB 8 BPP encoded
+set of PNGs in the working directory under a directory named "test_pngs" with files named
+{0..1}.png. If these images do not exist, they will error out and the benchmark will move
+on to the next set of benchmarks.
```

Comment:
Probably. Does benchmark have a skip without error? If so I can use that.

Reply:
There is a `GTEST_SKIP()`.. but that is for google test..

Reply:
Looks as though there's only a "SkipWithError":
https://github.com/google/benchmark/blob/d08e7b60564549aeaaa23d7bb075b6b9abf01d49/src/benchmark.cc#L212

The error doesn't halt the rest of the benchmarks so I think it's probably the cleanest thing that we can do, since the bootstrapping of these tests is basically done at compile time.